rsp_jupyter_extensions/handlers/query.py

```diff
@@ -1,112 +1,288 @@
-"""
-This is a Handler Module to provide an endpoint for templated queries.
-"""
+"""Handler Module to provide an endpoint for templated queries."""
 import json
 import os
 from pathlib import Path
 
-from ._rspclient import RSPClient
-
 import tornado
+import xmltodict
+from jupyter_server.base.handlers import APIHandler
+from lsst.rsp import get_query_history
 
-from jupyter_server.base.handlers import JupyterHandler
-
-
-class UnsupportedQueryTypeError(Exception):
-    pass
-
-
-class UnimplementedQueryResolutionError(Exception):
-    pass
+from ..models.query import (
+    TAPQuery,
+    UnimplementedQueryResolutionError,
+    UnsupportedQueryTypeError,
+)
+from ._rspclient import RSPClient
+from ._utils import _peel_route, _write_notebook_response
 
 
-class QueryHandler(JupyterHandler):
-    """
-    RSP templated Query Handler.
-    """
+class QueryHandler(APIHandler):
+    """RSP templated Query Handler."""
 
     def initialize(self) -> None:
+        """Get a client to talk to Times Square and TAP APIs."""
         super().initialize()
-        self._client = RSPClient(base_path="times-square/api/v1/")
+        self._ts_client = RSPClient(base_path="times-square/api/v1/")
+        self._tap_client = RSPClient(base_path="/api/tap/")
+        self._cachefile = (Path(os.getenv("JUPYTER_SERVER_ROOT", "")) /
+                           ".cache" / "queries.json" )
```

Comment:
Since you  get this env var a vew times, perhaps we could store it here

```suggestion
        self._root_path = Path(os.getenv("JUPYTER_SERVER_ROOT", ""))
        self._cachefile = self._root_path / ".cache" / "queries.json"
```